In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('../data/zomato_dataset.csv', encoding='utf-8')
print("Dataset loaded successfully!")
print(f"Shape: {df.shape}")

Dataset loaded successfully!
Shape: (44891, 7)


In [3]:
df.head()

Restaurant Name Rating  \
0                        Campus Bakers    4.3   
1       Mama Chicken Mama Franky House      4   
2     GMB - Gopika Sweets & Restaurant    4.2   
3  Shree Bankey Bihari Misthan Bhandar    4.2   
4                          Burger King    4.2   

                                             Cuisine Average Price  \
0         Bakery, Fast Food, Pizza, Sandwich, Burger   ₹50 for one   
1        North Indian, Mughlai, Rolls, Burger, Momos   ₹50 for one   
2  North Indian, South Indian, Chinese, Fast Food...   ₹50 for one   
3  Mithai, Street Food, South Indian, Chinese, Ic...   ₹50 for one   
4                       Burger, Fast Food, Beverages   ₹50 for one   

  Average Delivery Time                                     Safety Measure  \
0                36 min            Restaurant partner follows WHO protocol   
1                22 min  Follows all Max Safety measures to ensure your...   
2                27 min  Follows all Max Safety measures to ensure your...   
3                28 min            Restaurant partner follows WHO protocol   
4                26 min  Follows all Max Safety measures to ensure your...   

  Location  
0     Agra  
1     Agra  
2     Agra  
3     Agra  
4     Agra

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44891 entries, 0 to 44890
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Restaurant Name        44891 non-null  object
 1   Rating                 44891 non-null  object
 2   Cuisine                44872 non-null  object
 3   Average Price          44891 non-null  object
 4   Average Delivery Time  44891 non-null  object
 5   Safety Measure         44891 non-null  object
 6   Location               44891 non-null  object
dtypes: object(7)
memory usage: 2.4+ MB


In [5]:
df.describe(include='all')

Restaurant Name Rating       Cuisine  \
count                                 44891  44891         44872   
unique                                35688     30         17580   
top     Baskin Robbins - Ice Cream Desserts    4.0  North Indian   
freq                                     70   5575          2403   

       Average Price Average Delivery Time  \
count          44891                 44891   
unique            28                   398   
top     ₹100 for one                30 min   
freq           13019                  1715   

                                           Safety Measure Location  
count                                               44891    44891  
unique                                                  2       98  
top     Follows all Max Safety measures to ensure your...  Kolkata  
freq                                                25726     1296

In [6]:
missing = df.isnull().sum()
missing_pct = (missing / len(df)) * 100
missing_df = pd.DataFrame({'Missing Count': missing, 'Percentage': missing_pct})
missing_df[missing_df['Missing Count'] > 0].sort_values('Percentage', ascending=False)

Missing Count  Percentage
Cuisine             19    0.042325

In [7]:
print(f"Total unique cities: {df['Location'].nunique()}")
print(df['Location'].unique())

Total unique cities: 98
['Agra' 'Ahmedabad' 'Ajmer' 'Alappuzha' 'Aligarh' 'Allahabad' 'Amravati'
 'Amritsar' 'Asansol' 'Aurangabad' 'Bangalore' 'Bareilly' 'Belgaum'
 'Bhavnagar' 'Bhopal' 'Bhubaneswar' 'Bikaner' 'Chandigarh' 'Chennai'
 'Coimbatore' 'Cuttack' 'Darbhanga' 'Darjeeling' 'Dehradun' 'Delhi NCR'
 'Dhanbad' 'Dharamshala' 'Durgapur' 'Firozabad' 'Gaya' 'Goa' 'Gonda'
 'Gorakhpur' 'Guntur' 'Guwahati' 'Haridwar' 'Hyderabad' 'Indore'
 'Jabalpur' 'Jaipur' 'Jalandhar' 'Jalgaon' 'Jammu' 'Jamnagar' 'Jamshedpur'
 'Jhansi' 'Jodhpur' 'Junagadh' 'Kanpur' 'Kharagpur' 'Kolhapur' 'Kolkata'
 'Kota' 'Lucknow' 'Ludhiana' 'Madurai' 'Manali' 'Mangalore' 'Manipal'
 'Meerut' 'Moradabad' 'Motihari' 'Mumbai' 'Mussoorie' 'Mysore' 'Nagpur'
 'Nainital' 'Nashik' 'NCR' 'Nellore' 'Palakkad' 'Patiala' 'Patna'
 'Puducherry' 'Pune' 'Raipur' 'Rajkot' 'Ranchi' 'Rishikesh' 'Saharanpur'
 'Salem' 'Sangli' 'Shimla' 'Siliguri' 'Solapur' 'Surat' 'Thrissur'
 'Tirunelveli' 'Trichy' 'Trivandrum' 'Udaipur' 'Ujjain' 'Vadodar

- Dataset has 44,891 rows and 7 columns
- All 7 columns are useful, nothing to drop
- Key columns: Restaurant Name, Rating, Cuisine, Average Price, Average Delivery Time, Safety Measure, Location
- Columns with missing data: Cuisine (19 missing, ~0.04%)
- All columns stored as strings — need type conversion in cleaning
- Dataset covers 98 cities across India